lib/ansible/modules/cloud/amazon/ecs_taskdefinition_facts.py

```diff
@@ -0,0 +1,346 @@
+            type: string
+        options:
+            description: The configuration options to send to the log driver.
+            returned: when present
+            type: string
+family:
+    description: The family of your task definition, used as the definition name
+    returned: always
+    type: string
+taskDefinitionArn:
+    description: ARN of the task definition
+    returned: always
+    type: string
+taskRoleArn:
+    description: The ARN of the IAM role that containers in this task can assume
+    returned: when role is set
+    type: string
+networkMode:
+    description: Network mode for the containers
+    returned: always
+    type: string
+revision:
+    description: Revision number that was queried
+    returned: always
+    type: int
+volumes:
+    description: The list of volumes in a task
+    returned: always
+    type: complex
+    contains:
+        name:
+            description: The name of the volume.
+            returned: when present
+            type: string
+        host:
+            description: The contents of the host parameter determine whether your data volume persists on the host container instance and where it is stored.
+            returned: when present
+            type: bool
+        sourcePath:
+            description: The path on the host container instance that is presented to the container.
+            returned: when present
+            type: string
+status:
+    description: The status of the task definition
+    returned: always
+    type: string
+requiresAttributes:
+    description: The container instance attributes required by your task
+    returned: when present
+    type: complex
+    contains:
+        name:
+            description: The name of the attribute.
+            returned: when present
+            type: string
+        value:
+            description: The value of the attribute.
+            returned: when present
+            type: string
+        targetType:
+            description: The type of the target with which to attach the attribute.
+            returned: when present
+            type: string
+        targetId:
+            description: The ID of the target.
+            returned: when present
+            type: string
+placementConstraints:
+    description: A list of placement constraint objects to use for tasks
+    returned: always
+    type: complex
+    contains:
+        type:
+            description: The type of constraint.
+            returned: when present
+            type: string
+        expression:
+            description: A cluster query language expression to apply to the constraint.
+            returned: when present
+            type: string
+'''  # NOQA
+
+try:
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import boto3_conn, ec2_argument_spec, get_aws_connection_info
+
+
+class EcsTaskManager:
```

Comment:
This class isn't really providing any value. Usual practice is to create the client in main and then call a describe_task_definitions method.

Reply:
Agreed. This is my first module and I started with code from another that had a similar class, so I thought it was a pattern. Glad to be rid of it.